src/BPDual.jl

```diff
@@ -171,6 +178,8 @@ function bpdual(
     numtrim = 0
     nprodA = 0
     nprodAt = 0
+    cur_r_size = 0
```

Comment:
Is tyhere a situation in which the variables `cur_r_size` and `nact` different? My understanding is that the size of `R` always corresponds to the number of active variables, indicated by `nact`. If so, would it be possible to instead use `nact` everywhere?

Also, it seems that if a warm-start us used, then `cur_r_size` may be incorrect.